modules/openapi-generator/src/main/resources/Java/libraries/native/ApiClient.mustache

```diff
@@ -158,7 +158,7 @@ public class ApiClient {
   public ApiClient() {
     this.builder = createDefaultHttpClientBuilder();
     this.mapper = createDefaultObjectMapper();
-    updateBaseUri(getDefaultBaseUri());
+    updateBaseUri("{{{basePath}}}");
```

Comment:
The usage of `getDefaultBaseUri()` in the constructor is fine until a user overrides it and references any dynamically calculated value (static string would be fine). 

A proper fix would be to have a constructor that accepts a `basePath` and doesn't offer a default value (or leave this method as is, but remove it from the constructor and introduce a factory method that creates the ApiClient and invokes `updateBaseUri` right after). That would be a breaking change, though would address the problem.

This change is a compromise of sorts, where a default value is taken directly from the variable (which user can still override or call `updateBaseUri` later). The getter is simply a getter now and isn't invoked anywhere.

Reply:
> A proper fix would be to have a constructor that accepts a basePath

agreed. that's what the okhttp-gson client offers.

https://github.com/OpenAPITools/openapi-generator/blob/master/modules/openapi-generator/src/main/resources/Java/libraries/okhttp-gson/ApiClient.mustache#L211

and we can leave it to another PR instead